prover/src/lib.rs

```diff
@@ -456,17 +456,17 @@ pub trait Prover {
     /// building a Merkle tree from the resulting hashes.
     fn build_trace_commitment<E>(
         &self,
-        trace: &Matrix<E>,
+        trace: &ColMatrix<E>,
         domain: &StarkDomain<Self::BaseField>,
-    ) -> (Matrix<E>, MerkleTree<Self::HashFn>, Matrix<E>)
+    ) -> (RowMatrix<E>, MerkleTree<Self::HashFn>, ColMatrix<E>)
     where
         E: FieldElement<BaseField = Self::BaseField>,
     {
         // extend the execution trace
         #[cfg(feature = "std")]
         let now = Instant::now();
         let trace_polys = trace.interpolate_columns();
-        let trace_lde = trace_polys.evaluate_columns_over(domain);
+        let trace_lde = RowMatrix::evaluate_polys_over::<8>(&trace_polys, domain);
```

Comment:
I'm curious how the value 8 has been chosen, was it experimentally?

Reply:
Yes, this seems to work the best for 64-bit fields (at least on Apple M1). I'll create a constant for this and will add some comments.